sale_payment_mgmt/models/account_payment.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2020 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class AccountPayment(models.Model):
+    _inherit = "account.payment"
+
+    def post(self):
+        obj = self
+        if (
+            self.env.user.has_group("sales_team.group_sale_salesman")
+            or self.env.user.has_group("sales_team.group_sale_salesman_all_leads")
+            or self.env.user.has_group("sales_team.group_sale_manager")
+        ) and not self.env.user.has_group("account.group_account_invoice"):
+            obj = self.sudo()
+        return super(AccountPayment, obj).post()
+
+    @api.model
+    def fields_view_get(
+        self, view_id=None, view_type="form", toolbar=False, submenu=False
+    ):
+        result = super(AccountPayment, self).fields_view_get(
+            view_id, view_type, toolbar=toolbar, submenu=submenu,
+        )
+        if view_type == "form":
+            if (
+                self.env.user.has_group("sales_team.group_sale_salesman")
```

Comment:
The same. You can refactor anyway this into a method `_is_sale_payment_managed` or similar for calling it and getting True/False.